### PR TITLE
BaSP-MR-43: Class GET - GETBYID - POST

### PR DESCRIPTION
**Pull request by IgnacioGalceran**
Display of classes list (Get)
Create form for classes (Post)
Select of a class (GetById)
Search filter in the class list

### CODE REVIEW

File: src/Components/Classes/Form/Form.jsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    trainer: '',
+    day: '',
+    time: '',
+    capacity: ''
+  });
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+    if (
+      Object.values(classes).every((el) => {
```
Comment (josedaniele) [inline src/Components/Classes/Form/Form.jsx]: What is "el"? Could you give it a more representative name?

File: src/Components/Classes/Form/Form.jsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+                {error && classes.day === '' ? (
+                  <span className={styles.error}>Field is required</span>
+                ) : null}
+              </div>
+              <div className={styles.inputContainer}>
+                <label>Time</label>
+                <input type="time" value={classes.time} name="time" onChange={onChangeInput} />
+                {error && classes.time === '' ? (
+                  <span className={styles.error}>Field is required</span>
+                ) : null}
```
Comment (josedaniele) [inline src/Components/Classes/Form/Form.jsx]: can include an error condition where "classes.time === 0 "  to prevent making a server call.

**Review by josedaniele — CHANGES_REQUESTED** [review]

**Review by josedaniele — APPROVED** [review]

File: src/Components/Classes/index.jsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
+import styles from './classes.module.css';
+import ClassList from './List/ClassList';
+import Form from './Form/Form';
+import Modal from './Modal/Modal';
+import { useState, useEffect } from 'react';
```
Comment (JuanCruzMoreira) [inline src/Components/Classes/index.jsx]: Import React too
`import React, { useEffect, useState } from 'react';
`

**Review by JuanCruzMoreira — CHANGES_REQUESTED** [review]

**Review by JuanCruzMoreira — APPROVED** [review]

File: src/Components/Classes/List/list.module.css [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+tbody tr:nth-child(even):hover {
+  background-color: #c0b6b6;
+}
+
+.filter {
+  margin-bottom: 20px;
+}
+
+.selected {
+  background-color: #373867 !important;
```
Comment (JuanCruzMoreira) [inline src/Components/Classes/List/list.module.css]: Delete all !important in css files

**Review by JuanCruzMoreira — CHANGES_REQUESTED** [review]

File: src/Components/Classes/List/list.module.css (hunk 4 above)
Comment (IgnacioGalceran) [reply src/Components/Classes/List/list.module.css]: done!

**Review by JuanCruzMoreira — APPROVED** [review]

**Pull request merged**